sympy/core/basic.py

```diff
@@ -739,7 +702,9 @@ def expr_free_symbols(self):
             active_deprecations_target="deprecated-expr-free-symbols")
         return set()
 
-    def as_dummy(self):
+    # XXX: The as_dummy() return type should be Self but need Python >= 3.11
```

Comment:
There is some workaround to this
You can import `Self` conditionally from `typing_extensions`
```
if TYPE_CHECKING:
    from typing_extensions import Self
```
and use stringified annotation like `def as_dummy(self) -> "Self":`.

Reply:
Does that work if `typing_extensions` is not installed?

Reply:
We can try that, but I expect that may work as long as typing_extensions is added to the type checking CI job.

Reply:
Using `Self` causes the docs build to fail at least in circle CI. The cir cle CI build uses Python 3.10 although I'm not sure if that matters.

Reply:
I am not having error with sphinx built with python 3.11 and typing_extensions installed in that environment as well